geoutils/georaster/raster.py

```diff
@@ -2027,11 +2025,38 @@ def value_at_coords(
             (c = provided coordinate, v= value of surrounding coordinate)
 
         """
-        value: float | dict[int, float] | tuple[float | dict[int, float] | tuple[list[float], np.ndarray] | Any]
+        # Check for array-like inputs
+        if (
+            not isinstance(x, (float, np.floating, int, np.integer))
+            and isinstance(y, (float, np.floating, int, np.integer))
+            or isinstance(x, (float, np.floating, int, np.integer))
+            and not isinstance(y, (float, np.floating, int, np.integer))
+        ):
+            raise TypeError("Coordinates must be both numbers or both array-like.")
+
+        # If for a single value, wrap in a list
+        if isinstance(x, (float, np.floating, int, np.integer)) and isinstance(
```

Comment:
The second condition is redundant after the first check...

Reply:
Ah yes good catch, updated!